### PR TITLE
updated comment, blog and createpost section

### PR DESCRIPTION
**Pull request by Macbright007**

### CODE REVIEW

File: src/components/SinglePost.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    try {
+      const result = await Axios.post('https://blogpostapi1.herokuapp.com/comment', {
+        content,
+        post: params.id,
+      });
+      const newComment = result.data.data[0] //fetching new comment
+      const newCommentList = [...comments] //copying old comment with new one
+      newCommentList.push(newComment) //updating old comment with new one
+
+      setComments(newCommentList)
```
Comment (Asylcreek) [inline src/components/SinglePost.js:26]: You can use the `setState function arg` to achieve the same functionality

File: src/components/SinglePost.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const [content, setContent] = useState('');
+  const [comments, setComments] = useState([]);
+  const [post, setPost] = useState(null)
+
+  const params = useParams();
+
+  const handleClick = async (e) => {
+    e.preventDefault();
+    try {
+      const result = await Axios.post('https://blogpostapi1.herokuapp.com/comment', {
```
Comment (Asylcreek) [inline src/components/SinglePost.js]: You saved your url in a variable and you are not using it

File: src/components/SinglePost.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  };
+
+  useEffect(() => {
+    const getComments = async () => {
+      try {
+        //combinning all the request together with a promise
+        const [post, comments] = await Promise.all([
+          Axios.get(
+            `https://blogpostapi1.herokuapp.com/${params.id}`
```
Comment (Asylcreek) [inline src/components/SinglePost.js]: You saved your url in a variable and you are not using it

File: src/components/SinglePost.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          ),
+          Axios.get(
+            `https://blogpostapi1.herokuapp.com/comment/all_comments/${params.id}`
```
Comment (Asylcreek) [inline src/components/SinglePost.js]: You saved your url in a variable and you are not using it

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/SinglePost.js (hunk 1 above)
Comment (Macbright007) [reply src/components/SinglePost.js:26]: i dont get this please

File: src/components/SinglePost.js (hunk 2 above)
Comment (Macbright007) [reply src/components/SinglePost.js]: okay

File: src/components/SinglePost.js (hunk 1 above)
Comment (Asylcreek) [reply src/components/SinglePost.js:26]: `setComment((prev) => ([...prev, newComment]))`

**Review by Asylcreek — APPROVED** [review]

File: src/components/SinglePost.js (hunk 1 above)
Comment (Macbright007) [reply src/components/SinglePost.js:26]: on line 22 or 23??

**Comment by Macbright007** [comment]
can i merge them now?

File: src/components/SinglePost.js (hunk 1 above)
Comment (Asylcreek) [reply src/components/SinglePost.js:26]: Lines 21 - 25

**Comment by Asylcreek** [comment]
Sure.

**Pull request merged**